encrypted-media/scripts/requestmediakeysystemaccess.js

```diff
@@ -0,0 +1,311 @@
+function runTest( config, qualifier ) {
+
+    var testname = testnamePrefix( qualifier, config.keysystem ) + ', requestMediaKeySystemAccess';
+
+    function expect_error(keySystem, configurations, expectedError, testname) {
+        promise_test(function(test) {
+            return navigator.requestMediaKeySystemAccess(keySystem, configurations).then(function(a) {
+                assert_unreached('Unexpected requestMediaKeySystemAccess() success.');
+            }, function(e) {
+                assert_equals(e.name, expectedError);
+            });
+        }, testname);
+    }
+
+    function assert_subset(actual, expected, path) {
```

Comment:
It's comparing an expected value of _configuration_ with the actual value, to check that the expected value is a "subset" of the actual value. 

For strings, we expect equality, for arrays we expect equal length and corresponding elements to have the subset property. For objects, we expect every member of the expected configuration to be present in the tested value and have the same subset property.

So, there may be other elements in objects within the actual value which we ignore.

There is nothing defined in the configuration object that is not an object, array or string, so we do not need to consider other data types.